### ISSUE TITLE
inaccurate behaviour of find references

### ISSUE THREAD

**Issue by nulld** [vshaxe/vshaxe#470]
`ConsiderDemandRewardChoice` and `RandomRewardChoice` both implements the same interface:
```
interface IChooseRewardStrategy
{
	function forDailyQuest(template:TMtRewardTemplate):TCost;
	function forMarket(template:TMtRewardTemplate):TCost;
	function forCamping(rewardLength:Int, choiceSet:Array<Int>):TCost;
	function forZeppelin(rewardLength:Int, choiceSet:Array<Int>, weights:Array<Float>):TCost;
	function forDropPack(pack:DropPack):TCost;
	function getName():String;
}
```

Expected behaviour: 
* only one reference (or `0` references - not sure how it will be correct) to `ConsiderDemandRewardChoice.forCamping` from `IChooseRewardStrategy.forCamping` (it not direct accessed, only via inststance of `IChooseRewardStrategy` type). (see screen1)
* references to `IChooseRewardStrategy.forCamping` where `forCamping` accessed via interface. (see screen2)

**screen1**
![image](https://user-images.githubusercontent.com/2820168/101374752-9e428280-38bf-11eb-90f1-3eda13d9ea16.png)

**screen2**
![image](https://user-images.githubusercontent.com/2820168/101376524-bfa46e00-38c1-11eb-9ff0-81175e7d9ddb.png)

**Comment by nulld** [comment]
```
"name": "vshaxe",
"version": "2.21.4",
```
hope this little sample helps to reproduce the problem:
[issue470.zip](https://github.com/vshaxe/vshaxe/files/5655182/issue470.zip)

**Comment by nadako** [comment]
It's a bit hard to understand the actual problem and reproduction steps, could you describe that using the class/interface/method names from the code sample from that zip?

**Comment by nulld** [comment]
Hope this picture helps to understand the issue :)
![image](https://user-images.githubusercontent.com/2820168/101407601-2b9bcc00-38ec-11eb-912d-e95cf71c5d91.png)

**Comment by nadako** [comment]
This is actually intended, see discussions in https://github.com/HaxeFoundation/haxe/issues/9044, https://github.com/HaxeFoundation/haxe/pull/9315, https://github.com/HaxeFoundation/haxe/issues/9470, https://github.com/HaxeFoundation/haxe/issues/9503 and probably dozen others :)

TLDR: it's what you most often actually want, but not always. Haxe server supports specifying options that configure what to include, but vshaxe currently only uses the most inclusive one, because there's no easy UI to configure this in VS Code.

**Comment by nulld** [comment]
Got it, and look to this case please:
Is it considered as correct too?
![image](https://user-images.githubusercontent.com/2820168/101408442-75d17d00-38ed-11eb-8084-d4fad9546b8f.png)

**Comment by nadako** [comment]
I think this is what is/was discussed in https://github.com/HaxeFoundation/haxe/issues/9503

**Comment by nulld** [comment]
Ok, got it, thanks for sharing.
Will accept it like a feature not a bug :)
